src/getStats.ts

```diff
@@ -0,0 +1,25 @@
+import { DiagnosticCategory } from "typescript";
+
+import { DiagnosticError } from "./types";
+
+export type Stats = {
+  errors: number;
+  warnings: number;
+};
+
+export function getStats(diagnostics: DiagnosticError[]): Stats {
+  const stats: Stats = {
+    errors: 0,
+    warnings: 0,
+  };
+
+  diagnostics.forEach(({ category }) => {
```

Comment:
Just a convenience question, why forEach and mutate and not reduce over the stats?

Reply:
I don't know 😂 I'll change it 👌